File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {

    public String encode(List<String> strs) {
        String ans = "";

        for(String s : strs) ans += (s.length() + "#" + s);

        return ans;
    }

    public List<String> decode(String str) {
        int i = 0, j = str.length() - 1;
        
        List<String> ans = new ArrayList<>();

        while(i <= j){
            String s = "";

            String temp = "";
            while(str.charAt(i) != '#') temp += str.charAt(i++);

            int len = Integer.parseInt(temp);

            i++;

            int end = i + len;

            while(i < end) s += str.charAt(i++);

            ans.add(s);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-1.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s = "";
        string smallest = strs[0];

        for(string str : strs){
            if(str.size() < smallest.size()) smallest = str;
        }

        for(int i = 0; i < smallest.size(); i++){
            for(string str : strs){
                if(str[i] != smallest[i]) return s;
            }
            s += smallest[i];
        }

        return s;
    }
};